File: order-service/src/main/java/com/intexsoft/webshop/orderservice/model/Order.java
package com.intexsoft.webshop.orderservice.model;

import com.intexsoft.webshop.orderservice.model.enums.PaymentMethod;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.SourceType;
import org.hibernate.proxy.HibernateProxy;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.LinkedHashSet;
import java.util.Objects;
import java.util.Set;

@Entity
@Table(name = "\"order\"")
@Getter
@Setter
@ToString
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;
    @Column(name = "pickup_point_id", nullable = false)
    Long pickupPointId;
    @Column(name = "shop_id", nullable = false)
    Long shopId;
    @Column(name = "user_id", nullable = false)
    Long userId;
    @CreationTimestamp(source = SourceType.DB)
    @Column(name = "created_at", nullable = false)
    LocalDateTime createdAt;
    @Column(name = "total_price", nullable = false)
    BigDecimal totalPrice;
    @Column(name = "payment_method", nullable = false, length = 15)
    @Enumerated(EnumType.STRING)
    PaymentMethod paymentMethod;
    @Column(name = "comment")
    String comment;
    @OneToMany(
            mappedBy = "order",
            cascade = CascadeType.ALL,
            fetch = FetchType.LAZY,
            orphanRemoval = true
    )
    @ToString.Exclude
    Set<Detail> orderDetails = new LinkedHashSet<>();
    @OneToMany(
            mappedBy = "order",
            cascade = CascadeType.ALL,
            fetch = FetchType.LAZY,
            orphanRemoval = true
    )
    @ToString.Exclude
    Set<Status> statuses = new LinkedHashSet<>();

    public void addDetail(Detail detail) {
        orderDetails.add(detail);
        detail.setOrder(this);
    }

    public void removeDetail(Detail detail) {
        orderDetails.remove(detail);
        detail.setOrder(null);
    }

    public void addStatus(Status status) {
        statuses.add(status);
        status.setOrder(this);
    }

    public void removeStatus(Status status) {
        statuses.remove(status);
        status.setOrder(this);
    }

    @Override
    public final boolean equals(Object object) {
        if (this == object) return true;
        if (object == null) return false;
        Class<?> oEffectiveClass = object instanceof HibernateProxy ? ((HibernateProxy) object).getHibernateLazyInitializer().getPersistentClass() : object.getClass();
        Class<?> thisEffectiveClass = this instanceof HibernateProxy ? ((HibernateProxy) this).getHibernateLazyInitializer().getPersistentClass() : this.getClass();
        if (thisEffectiveClass != oEffectiveClass) return false;
        Order order = (Order) object;
        return getId() != null && Objects.equals(getId(), order.getId());
    }

    @Override
    public final int hashCode() {
        return this instanceof HibernateProxy ? ((HibernateProxy) this).getHibernateLazyInitializer().getPersistentClass().hashCode() : getClass().hashCode();
    }
}

File: order-service-kt/src/main/kotlin/com/intexsoft/webshop/orderservicekt/model/Order.kt
package com.intexsoft.webshop.orderservicekt.model

import com.intexsoft.webshop.orderservicekt.model.enums.PaymentMethod
import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.SourceType
import java.math.BigDecimal
import java.time.LocalDateTime

@Entity
@Table(name = "\"order\"")
class Order(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0,
    @Column(name = "pickup_point_id", nullable = false)
    var pickupPointId: Long,
    @Column(name = "shop_id", nullable = false)
    var shopId: Long,
    @Column(name = "user_id", nullable = false)
    var userId: Long,
    @CreationTimestamp(source = SourceType.DB)
    @Column(name = "created_at", nullable = false)
    var createdAt: LocalDateTime,
    @Column(name = "total_price", nullable = false)
    var totalPrice: BigDecimal,
    @Column(name = "payment_method", nullable = false, length = 15)
    @Enumerated(EnumType.STRING)
    var paymentMethod: PaymentMethod,
    @Column(name = "comment")
    var comment: String?,
    @OneToMany(mappedBy = "order", cascade = [CascadeType.ALL], fetch = FetchType.LAZY, orphanRemoval = true)
    var orderDetails: MutableSet<Detail> = LinkedHashSet(),
    @OneToMany(mappedBy = "order", cascade = [CascadeType.ALL], fetch = FetchType.LAZY, orphanRemoval = true)
    var statuses: MutableSet<Status> = LinkedHashSet()

) {
    fun addDetail(detail: Detail) {
        orderDetails.add(detail)
        detail.order = this
    }

    fun removeDetail(detail: Detail) {
        orderDetails.remove(detail)
        detail.order = null
    }

    fun addStatus(status: Status) {
        statuses.add(status)
        status.order = this
    }

    fun removeStatus(status: Status) {
        statuses.remove(status)
        status.order = null
    }
}